lib/propolis-client/src/handmade/api.rs

```diff
@@ -186,6 +186,49 @@ pub struct Instance {
     pub nics: Vec<NetworkInterface>,
 }
 
+/// Request a specific range of an Instance's serial console output history.
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema, PartialEq)]
+pub struct InstanceSerialConsoleHistoryRequest {
+    /// Character index in the serial buffer from which to read, counting the bytes output since
+    /// instance start. If this is not provided, `most_recent` must be provided, and if this *is*
```

Comment:
Can/should this be made into an option-of-enum to avoid the caller-supplied-both/caller-supplied-neither cases entirely? (Or will that run into issues with expressing this type in the generated API spec? ISTR hitting a problem along those lines once; it was with a different type but it made clear to me that I don't know all the rules here.)

Reply:
i originally did that in my first pass of this feature in omicron, but i believe it shook out that rusty tagged enums is awkward for non-rust openapi consumers

Reply:
Makes sense, thanks!